engine/src/main/java/io/camunda/zeebe/engine/processing/bpmn/event/IntermediateThrowEventProcessor.java

```diff
@@ -200,15 +203,24 @@ public boolean isSuitableForEvent(final ExecutableIntermediateThrowEvent element
     @Override
     public void onActivate(
         final ExecutableIntermediateThrowEvent element, final BpmnElementContext activating) {
-
-      final var activated = stateTransitionBehavior.transitionToActivated(activating);
-      final boolean canBeCompleted =
-          eventPublicationBehavior.throwEscalationEvent(
-              element.getId(), element.getEscalation(), activated);
-
-      if (canBeCompleted) {
-        stateTransitionBehavior.completeElement(activated);
-      }
+      expressionProcessor
+          .evaluateStringExpressionAsDirectBuffer(
+              element.getEscalation().getEscalationCodeExpression(),
+              activating.getElementInstanceKey())
+          .ifRightOrLeft(
+              escalationCode -> {
+                final var escalation = element.getEscalation();
+                escalation.setEscalationCode(escalationCode);
```

Comment:
❌ We shouldn't change the model by setting the `escalationCode` here. We should move the evaluation of this expression to the `BpmnEventPublicationBehavior.throwEscalationEvent` method. This also removes the duplication from this processor and the end event processor.